bam_to_mate_hist.py

```diff
@@ -11,147 +11,291 @@
+    print stat_list[0], "of total", num_pairs, ", fraction ", stat_dict["NUM_DIFF_CONTIG_PAIRS"]
+
+    stat_dict["NUM_SPLIT_READS"] = "{0:.3f}".format(stat_list[3] / float(num_pairs * 2))
+    print "Count of split reads (more is usually good, as indicates presence of Hi-C junction in read):"
+    print stat_list[3], "of total", num_pairs * 2, ", fraction ", stat_dict["NUM_SPLIT_READS"]
+
+    stat_dict["NUM_DUPE_READS"] = "{0:.3f}".format(stat_list[4] / float(num_pairs * 2))
+    print "Count of duplicate reads (duplicates are bad; WILL ALWAYS BE ZERO UNLESS BAM FILE IS PREPROCESSED TO SET THE DUPLICATES FLAG):"
+    print stat_list[4], "of total", num_pairs * 2, ", fraction ", stat_dict["NUM_DUPE_READS"]
+
+    if count_diff_refname_stub:
+        print "Count of read pairs with mates mapping to different reference groupings, e.g. genomes (sign of bad " \
+              "prep potentially):"
+        print stat_list[2], "of total", num_pairs, ", fraction", float(stat_list[2]) / num_pairs
+
+    return stat_dict
+
 
 if __name__ == "__main__":
-	c_args = parse_args(__file__)
-	num_reads = int(c_args["num_reads"])
-	bamfile_handle = c_args["bam_file"]
-	outpdf_name = c_args["outpdf_name"]
-	count_diff_refname_stub = c_args["count_diff_refname_stub"]
-	print "parsing the first {0} reads in bam file {1} to QC Hi-C library quality, plots"\
-	" are written to {2}".format(
-		num_reads, bamfile_handle, outpdf_name
-		)
-	diff_chr, dists, diff_stub, split_reads = parse_bam_file(num_reads=num_reads, bamfile_handle=bamfile_handle, 
-		count_diff_refname_stub=count_diff_refname_stub)
-	print "Counts of zero distances (many is a sign of bad prep):"
-	unique, counts = np.unique(dists, return_counts=True)
-	#print unique[-100:-1]
-	zero_dist = dict(zip(unique, counts))[0]
-	print zero_dist, "of total", len(dists), "fraction ", float(zero_dist) / len(dists)
-	above_10k = len([dist for dist in dists if dist > 10000])
-	print "Count of read pairs with distance > 10KB (many is a sign of good prep):"
-	print above_10k, "of total", len(dists), ", fraction ", float(above_10k) / len(dists)
-	print "Count of read pairs with mates mapping to different chromosomes/contigs (sign of good prep IF same genome):"
-	print diff_chr, "of total", len(dists), ", fraction ", float(diff_chr) / len(dists)
-	print "Count of split reads (more is usually good, as indicates presence of Hi-C junction in read):"
-	print split_reads, "of total", len(dists)*2, ", fraction ", split_reads / float(len(dists)*2)
-	
-	if count_diff_refname_stub:
-		print "Count of read pairs with mates mapping to different reference groupings, e.g. genomes (sign of bad prep potentially):"
-		print diff_stub, "of total", len(dists), ", fraction", float(diff_stub) / len(dists)
-		
-	make_histograms(dists=dists, bamfile_handle=bamfile_handle, num_reads=num_reads, outpdf_name=outpdf_name)
-	
+    c_args = parse_args(__file__)
+    num_reads = int(c_args["num_reads"])
```

Comment:
If you set the type in your argument parser argument for num_reads, it will return an int so you don't need to do a conversion.  Not blocking, just a note for the future.